src/WebJobs.Script.WebHost/Management/LinuxSpecialization/RunFromPackageHandler.cs

```diff
@@ -0,0 +1,280 @@
+                    }
+                }
+                else
+                {
+                    await _meshServiceClient.MountFuse(MeshServiceClient.SquashFsOperation, filePath, scriptPath);
+                }
+            }
+            else if (packageType == CodePackageType.Zip)
+            {
+                // default to unzip for zip packages
+                if (_environment.IsMountEnabled())
+                {
+                    await _meshServiceClient.MountFuse(MeshServiceClient.ZipOperation, filePath, scriptPath);
+                }
+                else
+                {
+                    if (useLocalSitePackages)
+                    {
+                        _bashCommandHandler.UnzipPackage(filePath, localSitePackagesPath);
+                        await CreateBindMount(localSitePackagesPath, scriptPath);
+                    }
+                    else
+                    {
+                        _bashCommandHandler.UnzipPackage(filePath, scriptPath);
+                    }
+                }
+            }
+        }
+
+        private CodePackageType GetPackageType(string filePath, RunFromPackageContext pkgContext)
+        {
+            // cloud build always builds squashfs
+            if (pkgContext.IsScmRunFromPackage())
+            {
+                return CodePackageType.Squashfs;
+            }
+
+            var uri = new Uri(pkgContext.Url);
+            // check file name since it'll be faster than running `file`
+            if (FileIsAny(".squashfs", ".sfs", ".sqsh", ".img", ".fs"))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (FileIsAny(".zip"))
+            {
+                return CodePackageType.Zip;
+            }
+
+            // Check file magic-number using `file` command.
+            (var output, _, _) = _bashCommandHandler.RunBashCommand($"{BashCommandHandler.FileCommand} -b {filePath}", MetricEventNames.LinuxContainerSpecializationFileCommand);
+            if (output.StartsWith("Squashfs", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Squashfs;
+            }
+            else if (output.StartsWith("Zip", StringComparison.OrdinalIgnoreCase))
+            {
+                return CodePackageType.Zip;
+            }
+            else
+            {
+                throw new InvalidOperationException($"Can't find CodePackageType to match {filePath}");
+            }
+
+            bool FileIsAny(params string[] options)
+                => options.Any(o => uri.AbsolutePath.EndsWith(o, StringComparison.OrdinalIgnoreCase));
+        }
+
+        public async Task<string> Download(RunFromPackageContext pkgContext)
+        {
+            var zipUri = new Uri(pkgContext.Url);
+            if (!Utility.TryCleanUrl(zipUri.AbsoluteUri, out string cleanedUrl))
+            {
+                throw new Exception("Invalid url for the package");
```

Comment:
InvalidOperationException ?

Reply:
Dont have a preference :) switching to InvalidOperationException